include/executors/executor_base.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <argument_parsers/copy_argument_parser.h>
+
+class ExecutorBase {
+public:
+    ExecutorBase(CopyArgumentParser args);
```

Comment:
The API is incorrect. 
Please, don't use raw pointers in the modern C++.
Prefer either by value or by reference or by smart ptr.